fdbclient/include/fdbclient/BlobCipher.h

```diff
@@ -181,6 +192,186 @@ struct hash<BlobCipherDetails> {
 };
 } // namespace std
 
+#pragma pack(push, 1) // exact fit - no padding
+
+// Why BinarySerialization instead of ObjectSerialization?
+//
+// By running experiments comparing: BlobCipherEncryptHeaderFlagsV1 and AesCtr algorithms header (with/without
+// authentication encryption), below table summarizes on-disk storage penalty (bytes) due to storing
+// BlobCipherEncryptHeader using both formats:
+//
+//  ----------------------------------------------------------------------------------------------------------
+// |         S.No      |   ObjFlags  |  BinaryFlags |  ObjectAlgo  | BinaryAlgo | TotalObject  | TotalBinary  |
+// | ----------------- | ----------- | ------------ |  ----------- | ---------- | ------------ | ------------ |
+// |   AesCtrNoAuth    |     40      |     3        |       104    |      40    |     208      |     55       |
+// |   AesCtrHmacSha   |     40      |     3        |       184    |      96    |     288      |    111       |
+// |   AesCtrAesCmac   |     40      |     3        |       168    |      80    |     272      |     95       |
+//  ----------------------------------------------------------------------------------------------------------
+
+struct BlobCipherEncryptHeaderFlagsV1 {
+	// Serializable fields
+
+	uint8_t encryptMode;
+	uint8_t authTokenMode;
+	uint8_t authTokenAlgo;
+
+	bool operator==(const BlobCipherEncryptHeaderFlagsV1& o) const {
+		return encryptMode == o.encryptMode && authTokenMode == o.authTokenMode && authTokenAlgo == o.authTokenAlgo;
+	}
+
+	static StringRef toStringRef(const BlobCipherEncryptHeaderFlagsV1& flags, Arena& arena) {
+		BinaryWriter wr(AssumeVersion(ProtocolVersion::withEncryptionAtRest()));
+		wr.serializeBytes(&flags, sizeof(BlobCipherEncryptHeaderFlagsV1));
+		return wr.toValue(arena);
+	}
+
+	static BlobCipherEncryptHeaderFlagsV1 fromStringRef(const StringRef& flagsRef) {
```

Comment:
looks like this is redundant as we can do `BinaryReader::fromStringRef<BlobCipherEncryptHeaderFlagsV1>(flagsRef)`?